proposals/15-message-sending-ux-in-lotus.md

```diff
@@ -0,0 +1,135 @@
+
+Some of them are:
+- sending messages with unsynced node
+- sending messages with too low fees given current network conditions
+- having too little funds to send given message
+- head of the line blocking on previous messages
+
+Many of the issues that users hit while trying to interact with the chain stem from the fact that Lotus CLI is lacking interactive feedback. The CLI does not warn the user when these situations happen, and if it does the only resolution available right now is failing (and expose flag like `--force` for advanced use-cases). This approach only increases user confusion as its not clear for them how to proceed.
+
+The aim of the project is to make actively interacting with Filecoin network through Lotus a delightful experience by removing the multitude of sharp corners awaiting the user.
+
+
+#### Assumptions & hypotheses
+<!--_What must be true for this project to matter?_-->
+<!--(bullet list)-->
+- Lotus is used directly to interact with the chain. This will be the case until applications that fulfill roles of wallets, storage clients, retrieval clients are built. Even still application developers will likely use Lotus directly. 
+- Sharp edges of Lotus in this area are causing users to abandon/reduce their involvement.
+- 
+#### User workflow example
+<!--How would a developer or user use this new capability?\_-->
+<!--(short paragraph)-->
+
+_All of the below are reported issues with message sending UX._
+
+- Sending message from (partially) unsynced node, resulting in wrongly estimated gas fees.
+	- **Solution**: warn the user that node is not in sync, ask for confirmation to send the message
+- Default `MaxFee` configuration option results in message fees being too low, resulting in message not landing on chain:
+	- **Solution**: for interactive usage use the `MaxFee` as guideline. If message wouldn't immediately land on chain due to it (or would be at a risk of being squeezed out in short time), suggest to the user alternative fee. The alternative fee can be accepted, refused or modified. If low fee is specified, estimate the time message will need to land.
+- Previous messages are blocking new messages due to changing network conditions or insufficient funds.
+	- **Solution part 1**:  Warn users when new messages are submitted and ask for confirmation that user wants to proceed.
+	- **Solution part 2**: Build an interactive tool for observing state of messages, allowing the user to reprice them, discard them and showing the user what is the exact issue with them (head of the line blocking, insufficient funds, too low fee)
+
+#### Impact
+<!--\_How directly important is the outcome to web3 dev stack product-market fit?\_-->
+
+🔥🔥 
+This project has potential to decrease attrition of new users that are starting to experiment with Filecoin, while at the same time improving the experience of existing users. 
```

Comment:
Is this only relevant for lotus nodes that have sync’d the whole chain or also folks who want to do more lightweight message sending?

Reply:
You don't have to sync the whole chain and also I would also imagine some of the features could be supported by `lotus daemon --lite` without any additional work, where the `--lite` doesn't require any syncing at all.